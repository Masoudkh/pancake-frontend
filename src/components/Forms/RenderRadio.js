import React, {Fragment} from 'react';
import ErrorMessage from '../../components/Forms/Error';

var _ = require('lodash');

const isObject = (obj, key) => {
  return _.isObject(obj) ? obj[key] : obj;
};

const RenderRadio = fields => {
  const {label, isRequired, instructions, options, input} = fields;

  return <fieldset className={options && options.length > 2 ? 'radio-list' : 'radio-group'}>
    {label && <legend>
      {label}
      {isRequired && <span className="aria-hidden">(required)</span>}
    </legend>}
    {instructions && <p dangerouslySetInnerHTML={{ __html: instructions }}></p>}
    {!instructions && <Fragment></Fragment>}
    <div>
      <div>
        {options && options.map((item, key) => {
          return <label key={key} className={fields.className && fields.className}>
            <input {...input} type="radio" value={isObject(item, 'value')} />
            <span style={{border: '1px solid black'}}>{isObject(item, 'label')}</span>
          </label>;
        })}
      </div>
    </div>
    <ErrorMessage fields={fields} />
  </fieldset>;
};

export default RenderRadio;
