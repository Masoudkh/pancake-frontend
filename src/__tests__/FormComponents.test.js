import React from 'react';
import renderer from 'react-test-renderer';
import RenderRadio from '../components/Forms/RenderRadio';
import TextField from '../components/Forms/TextField';

describe('components', function() {



  describe('<RenderRadio />', function() {
    it('renders correctly', function() {
      let tree = renderer.create(<RenderRadio />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('<TextField />', function() {
    it('renders correctly', function() {
      let tree = renderer.create(<TextField />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

});
