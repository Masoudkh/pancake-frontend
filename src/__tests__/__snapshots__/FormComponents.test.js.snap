// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <RenderRadio /> renders correctly 1`] = `
<fieldset
  className="field radio-group"
>
  <div>
    <div />
  </div>
  <div />
</fieldset>
`;

exports[`components <TextField /> renders correctly 1`] = `
<fieldset
  className="field"
>
  <legend />
  <input
    type="text"
    value={undefined}
  />
  <div />
</fieldset>
`;
